File: core/src/backendServer/models/MachineGunPickup.js
const Pickup = require('./Pickup');
const constants = require('../settings/constants');
const ProjectileType = require('./ProjectileType');

class MachineGunPickup extends Pickup {
    constructor(x, y, id, ammunition = constants.MACHINE_GUN_BULLETS_IN_CHAMBER) {
        super(x, y, constants.MACHINE_GUN_SPRITE_WIDTH, constants.MACHINE_GUN_SPRITE_HEIGHT, ProjectileType.MACHINE_GUN,
            id, ammunition)
    }
}

module.exports = MachineGunPickup;

File: core/src/com/mygdx/game/model/pickup/MachineGunPickup.kt
package com.mygdx.game.model.pickup

import com.badlogic.gdx.graphics.Texture
import com.mygdx.game.model.pickup.Pickup
import com.mygdx.game.settings.MACHINE_GUN_SPRITE_HEIGHT
import com.mygdx.game.settings.MACHINE_GUN_SPRITE_WIDTH

class MachineGunPickup(
        x: Float = 0f,
        y: Float = 0f,
        texture: Texture) : Pickup(
        x,
        y,
        MACHINE_GUN_SPRITE_WIDTH,
        MACHINE_GUN_SPRITE_HEIGHT,
        texture
)